firebase-firestore/src/androidTest/java/com/google/firebase/firestore/QueryTest.java

```diff
@@ -1029,6 +1031,46 @@ public void testMultipleUpdatesWhileOffline() {
     assertEquals(asList(map("foo", "zzyzx", "bar", "2")), querySnapshotToValues(snapshot2));
   }
 
+  @Test
+  public void resumingQueryShouldRemoveDeletedDocumentsIndicatedByExistenceFilter() {
+    Map<String, Map<String, Object>> testDocs = new LinkedHashMap<>();
+    for (int i = 1; i <= 100; i++) {
+      testDocs.put("doc" + i, map("key", i));
+    }
+
+    // Setup firestore with disabled persistence and populate a collection with testDocs.
+    FirebaseFirestore firestore = testFirestore();
+    firestore.setFirestoreSettings(
+        new FirebaseFirestoreSettings.Builder().setPersistenceEnabled(false).build());
+    CollectionReference collection = firestore.collection(autoId());
+    writeAllDocs(collection, testDocs);
+
+    QuerySnapshot snapshot1 = waitFor(collection.get());
```

Comment:
Add a comment above this line that the query below will populate the cache and save the resume token.